Index: init.c
===================================================================
RCS file: /usr1/CVS/rtems/testsuites/sptests/spmkdir/init.c,v
retrieving revision 1.1
diff -u -r1.1 init.c
--- init.c	8 Jun 2010 13:22:58 -0000	1.1
+++ init.c	10 Aug 2010 09:19:44 -0000
@@ -64,6 +64,12 @@
   test_mkdir("a/b/c/4", S_IRWXO, 0);
   test_mkdir("a/b", omode, 0);
   test_mkdir("a", omode, 0);
+  test_mkdir("a/b/x", S_IRUSR, 0);
+  test_mkdir("a/b/x/y", S_IRUSR, -1);
+  
+  rv = mknod("a/n", S_IRWXU | S_IFREG, 0LL);
+
+  test_mkdir("a/n/b", S_IRUSR, -1);
 
   rv = open ("b", O_CREAT | O_RDONLY, omode);
   rtems_test_assert(rv >= 0);
